File: examples/python_modular/kernel_auc_modular.py
###########################################################################
# kernel can be used to maximize AUC instead of margin in SVMs 
###########################################################################

def auc ():
	print 'AUC'

	from shogun.Kernel import GaussianKernel, AUCKernel
	from shogun.Features import RealFeatures, Labels

	feats_train=RealFeatures(fm_train_real)
	width=1.7
	subkernel=GaussianKernel(feats_train, feats_train, width)

	kernel=AUCKernel(0, subkernel)
	kernel.setup_auc_maximization( Labels(label_train_real) )
	km_train=kernel.get_kernel_matrix()

if __name__=='__main__':
	from tools.load import LoadMatrix
	from numpy import double
	lm=LoadMatrix()
	fm_train_real=double(lm.load_numbers('../data/fm_train_real.dat'))
	label_train_real=lm.load_labels('../data/label_train_twoclass.dat')
	auc()

File: examples/undocumented/csharp_modular/kernel_auc_modular.cs
using System;

public class kernel_auc_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();
		double width = 1.6;

		double[,] train_real = Load.load_numbers("../data/fm_train_real.dat");
		double[] trainlab = Load.load_labels("../data/label_train_twoclass.dat");

		RealFeatures feats_train = new RealFeatures(train_real);
		GaussianKernel subkernel = new GaussianKernel(feats_train, feats_train, width);

		BinaryLabels labels = new BinaryLabels(trainlab);

		AUCKernel kernel = new AUCKernel(0, subkernel);
		kernel.setup_auc_maximization(labels);

		double[,] km_train = kernel.get_kernel_matrix();

		int numRows = km_train.GetLength(0);
		int numCols = km_train.GetLength(1);

		Console.Write("km_train:\n");

		for(int i = 0; i < numRows; i++){
			for(int j = 0; j < numCols; j++){
				Console.Write(km_train[i,j] +" ");
			}
			Console.Write("\n");
		}

	}
}
